smarts/core/agent_manager.py

```diff
@@ -223,25 +223,27 @@ def observe(
                 else:
                     logging.log(
                         logging.DEBUG,
-                        f"Agent `{agent_id}` has raised done with {observations[agent_id].events}",
+                        # pytype: disable=attribute-error
+                        f"Agent `{agent_id}` has raised done with {observations[agent_id].events}", 
+                        # pytype: enable=attribute-error
```

Comment:
This `attribute-error` is due to the two types of `observations` present. Consider fixing it as follows.
```suggestion
                obs, dones[agent_id] = Sensors.observe(
                    sim, agent_id, sensor_state, vehicle
                )
                observations[agent_id] = obs

                if sim.vehicle_index.vehicle_is_shadowed(vehicle.id):
                    # It is not a shadowing agent's fault if it is done
                    dones[agent_id] = False
                else:
                    logging.log(
                        logging.DEBUG,
                        f"Agent `{agent_id}` has raised done with {obs.events}", 
                    )
```